### PR TITLE
Include .nf-core.yml in `nf-core pipelines bump-version`

### PR DESCRIPTION
**Pull request by mashehu**

### CODE REVIEW

**Comment by mashehu** [comment]
@nf-core-bot changelog

File: nf_core/pipelines/bump_version.py [hunk 1]
```diff
@@ -161,70 +180,97 @@ def bump_nextflow_version(pipeline_obj: Pipeline, new_version: str) -> None:
...
         log.warning(f"File not found: '{fn}'")
         return
 
-    replacements = []
-    for pattern in patterns:
-        found_match = False
+    if yaml_key:
+        update_yaml_file(fn, patterns, yaml_key, required)
+    else:
+        update_text_file(fn, patterns, required)
```
Comment (mirpedrol) [inline nf_core/pipelines/bump_version.py:204]:
```suggestion
        update_text_file(fn=fn, patterns=patterns, required=required)
```

Comment (mashehu) [reply nf_core/pipelines/bump_version.py:204]: why only for `update_text_file`?

Comment (mirpedrol) [reply nf_core/pipelines/bump_version.py:204]: Ah sorry I read wrong! I thought it was `update_yaml_file()` and we were missing `yaml_key` 😄 You can ignore my suggestion then

File: nf_core/pipelines/bump_version.py [hunk 2]
```diff
@@ -161,70 +180,97 @@ def bump_nextflow_version(pipeline_obj: Pipeline, new_version: str) -> None:
...
+def update_file_version(
+    filename: Union[str, Path],
+    pipeline_obj: Pipeline,
+    patterns: List[Tuple[str, str]],
+    required: bool = True,
+    yaml_key: Optional[List[str]] = None,
+) -> None:
+    fn: Path = pipeline_obj._fp(filename)
 
-    Args:
```
[the remark this answers is not in the document]
Comment (mashehu) [reply nf_core/pipelines/bump_version.py:177]: done

**Review by mirpedrol — APPROVED** [review]
LGTM!

**Pull request merged**